obs2ioda-v2/src/f_c_string_t_mod.f90

```diff
@@ -0,0 +1,54 @@
+module f_c_string_t_mod
+    use iso_c_binding, only : c_loc, c_ptr, c_null_char, c_char, c_f_pointer, c_null_ptr
+    use f_c_string_i_mod, only : strlen
+    implicit none
+
+    type :: f_c_string_t
```

Comment:
I think more comments and documentation would be helpful. For example, if someone creates an instance of `f_c_string_t`, uses it to convert a Fortran string to a C string, and then destroys the `f_c_string_t` instance, is the C string still valid?